lua/ibl/init.lua

```diff
@@ -424,6 +535,50 @@ M.refresh = function(bufnr)
             inlay_hints.set(bufnr, row - 1, #whitespace, scope_hl.underline, scope_hl.underline)
         end
 
+        -- current indent start
+        if
+            current_indent_enabled
+            and config.current_indent.show_start
+            and is_current_indent_start
+            and not (
+                config.scope.enabled
+                and config.scope.show_start
+                and is_scope_start
+                and config.scope.priority >= config.current_indent.priority
+            )
+        then
+            vim.api.nvim_buf_set_extmark(bufnr, namespace, row - 1, #whitespace, {
+                end_col = #line,
+                hl_group = current_indent_hl.underline,
+                priority = config.current_indent.priority,
+                strict = false,
+            })
+            inlay_hints.set(bufnr, row - 1, #whitespace, current_indent_hl.underline, current_indent_hl.underline)
+        end
+
+        -- current indent end
+        if
+            current_indent_enabled
+            and config.current_indent.show_end
+            and is_current_indent_end
+            and #whitespace_tbl > current_indent_col_start_single
+            and not (
+                config.scope.enabled
+                and config.scope.show_end
+                and is_scope_end
+                and scope_show_end_cond
+                and config.scope.priority >= config.current_indent.priority
+            )
+        then
+            vim.api.nvim_buf_set_extmark(bufnr, namespace, row - 1, current_indent_col_start_single, {
```

Comment:
```suggestion
            vim.api.nvim_buf_set_extmark(bufnr, namespace, row - 1, #whitespace, {
```
We need to start highlights after the whitespace here (any highlights before the whitespace ends is already taken care of in the virtual text overlay). [Note: The code before the suggestion doesn't work with tabs since it depends on the `whitespace_tbl` instead of `whitespace`.]